KomaMRICore/src/simulation/Flow.jl

```diff
@@ -0,0 +1,44 @@
+function outflow_spin_reset!(args...; kwargs...)
+   return nothing
+end
+
+function outflow_spin_reset!(M, t, motion::MotionList; M0=0, seq_t=0, add_t0=false)
+   for m in motion.motions   
+      outflow_spin_reset!(M, t, m.action, m.time, m.spins; M0=M0, seq_t=seq_t, add_t0=add_t0)
+   end
+   return nothing
+end
+
+function outflow_spin_reset!(M, t, action::FlowPath, time, spins; M0=0, seq_t=0, add_t0=false)
+   t_unit = KomaMRIBase.unit_time(t, time)
+   idx = KomaMRIBase.get_indexing_range(spins)
+   M = @view(M[idx, :])
+   M0 = init_magnetization(M, M0)
+   t = init_time(t_unit, seq_t, add_t0)
+   itp = KomaMRIBase.interpolate(action.spin_reset, KomaMRIBase.Gridded(KomaMRIBase.Constant{KomaMRIBase.Previous}()), Val(size(action.spin_reset, 1)))
+   mask = KomaMRIBase.resample(itp, t)
+   mask .= (cumsum(mask; dims=2) .== 0)
+   mask_end = 1 .- vec(any(mask .== 0; dims=2))
+   if size(M, 2) > 1
+      M .*= mask
+      M .+= M0 .* (1 .- mask)
+   else
+      M .*= mask_end
+      M .+= M0 .* (1 .- mask_end)
+   end
+   return nothing
+end
```

Comment:
Please try to offer a solution instead of saying you don't know how to do something. I don't have a lot of time to think about this for you.

Conceptually, there shouldn't be any difference between the two cases, and the code needs to reflect that. If there is a difference, there is a design problem and you are mixing two cases together that should be separated, like applying this function to a `Mag` vs `AbstractArray`.

This whole code snippet needs to be simplified.
```julia
mask .= (cumsum(mask; dims=2) .== 0)
mask_end = 1 .- vec(any(mask .== 0; dims=2))
if size(M, 2) > 1
  M .*= mask
  M .+= M0 .* (1 .- mask)
else
  M .*= mask_end
  M .+= M0 .* (1 .- mask_end)
end
```
Regarding the `cumsum`,  instead, you can use `accumulate` with a boolean "or."

```julia
julia> A = (rand(3, 4) .> 0.85)
3×4 BitMatrix:
 0  0  1  0
 1  0  0  0
 0  0  0  1

julia> accumulate(|, A; dims=2)
3×4 BitMatrix:
 0  0  1  1
 1  1  1  1
 0  0  0  1
```

Reply:
Note that `reduce(+, ...)` and `accumulate(+, ...)` are the same as `sum` and `cumsum`.

Reply:
Except for the aspect of the dispatch between the magnetization and the vector, which can be applied, I have not been able to implement the rest of the changes. The accumulate function returns a bitmatrix that cannot be to handled on the GPU. Additionally, I tried to apply the approach of calculating t_reset, but for that, we need either the accumulate function we discussed or perform the interpolation.

So, although it may seem excesive, I don’t see a simpler option than interpolating, and at the same time, one that is compatible with both CPU and GPU.

Please keep in mind that I only turn to you after I have researched and tried to solve and address the problems on my own. The solution I end up committing is not the first one that comes to mind, but rather the result of many attempts and having tried it in various ways.

I will upload the suggested changes by removing the code snippet, but providing two definitions for the function instead.